enhancements/hide_container_mountpoints.md

```diff
@@ -0,0 +1,276 @@
+
+Parallel (Full output [here](https://raw.githubusercontent.com/lack/redhat-notes/main/crio_unshare_mounts/test_results/parallel.out))
+
+    Flaky tests:
+    
+    [k8s.io] [sig-node] Events should be sent by kubelets and the scheduler about pods scheduling and running  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
+    [sig-network] Conntrack should be able to preserve UDP traffic when server pod cycles for a ClusterIP service [Suite:openshift/conformance/parallel] [Suite:k8s]
+    
+    Failing tests:
+    
+    [k8s.io] [sig-node] Mount propagation should propagate mounts to the host [Suite:openshift/conformance/parallel] [Suite:k8s]
+    [sig-arch] Managed cluster should ensure control plane pods do not run in best-effort QoS [Suite:openshift/conformance/parallel]
+    [sig-network] Networking should provide Internet connection for containers [Feature:Networking-IPv4] [Skipped:azure] [Suite:openshift/conformance/parallel] [Suite:k8s]
+    
+    error: 5 fail, 951 pass, 1457 skip (20m27s)
+
+Serial (Full output [here](https://raw.githubusercontent.com/lack/redhat-notes/main/crio_unshare_mounts/test_results/serial.out))
+
+    Failing tests:
+    
+    [sig-auth][Feature:OpenShiftAuthorization][Serial] authorization  TestAuthorizationResourceAccessReview should succeed [Suite:openshift/conformance/serial]
+    [sig-cluster-lifecycle][Feature:Machines][Serial] Managed cluster should grow and decrease when scaling different machineSets simultaneously [Suite:openshift/conformance/serial]
+    
+    error: 2 fail, 58 pass, 227 skip (53m14s)
+
+The only significant risk here is that of the `Mount propagation should
+propagate mounts to the host` which is explictly and intentionally broken by
+this proposal.  See /Risks and Mitigations/ above for an in-depth description
+of the changes.
+
+## Drawbacks
+
+- Requires re-wrapping CRI-O and Kubelet services in bash and `nsenter`, and
+  this is a little fragile. Could be mitigated by altering CRI-O and Kubelet to
+  take the mount namespace PID and/or file on the command line. On the other
+  hand, wrapping via `nsenter` could be reused upstream for all compatible
+  container runtimes.
+- If the namespace service restarts and then either CRI-O or Kubelet restarts,
+  there will be a mismatch between the mount namespaces and containers will
+  start to fail. Could be mitigated by changing the namespace service to NOT
+  cleanup its pinned namespace but instead idempotently re-use a
+  previously-created namespace.  However, given that the namespace service as
+  implemented today is a systemd `oneshot` with no actual process that needs
+  keeping alive, the risk of this terminating unexpectedly is very low.
+
+## Alternatives
+
+- Enhance systemd to support unsharing namespaces at the slice level, then put
+  `crio.service` and `kubelet.service` in the same slice
+- Alter both CRI-O and Kubelet executables to take a mount namespace via
```

Comment:
I prefer that for CRI-O we support joining the mount namespace as an option. We should be able to make it work with an init in C similar to runc.

Reply:
The benefit to sticking with the 'nsenter' mechanism is that it would be the same for Kubelet, CRI-O, and easily adaptable to any other container runtime, but I am happy to consider natively supporting mount-namespace-entering for CRI-O as well.  It's just more work.